Add automated test framework support for KubeArmorHostPolicy

oneiro-naut opened the pull request:
This adds KubeArmorHostPolicy support to the existing test framework. To run the host test scenarios hostpolicy must be enabled in kubearmor using the `-enableHostPolicy` option.

The commits also add skipped and reran testcase information to be shown in the test results and during the test script run.

Fixes: #168

oneiro-naut committed:
Add automated test framework support for KubeArmorHostPolicy

Fixes: #168

Signed-off-by: Ayush Dwivedi <ayush.dwivedi@accuknox.com>

oneiro-naut committed:
Add KubeArmorHostPolicy support to test-scenarios-in-runtime.sh

Fixes: #168

Signed-off-by: Ayush Dwivedi <ayush.dwivedi@accuknox.com>

oneiro-naut committed:
Add reran info to test scripts and fix minor bugs in scripts

Signed-off-by: Ayush Dwivedi <ayush.dwivedi@accuknox.com>

nam-jaehyun commented on tests/test-scenarios-in-runtime.sh:
```diff
@@ -21,6 +21,11 @@ realpath() {
 
 TEST_HOME=`dirname $(realpath "$0")`
 
+SKIP_CONTAINER_POLICY=0
```
If we merge this script, it tests host policies by default.
So, let's set the flags for container policies by default.
Then, similar to -EnableHostPolicy, why don't we have some options (arguments) for host policy testing?

oneiro-naut replied:
It doesn't actually test the host policy and scenarios if we don't have the enableHostPolicy flag passed to kubearmor. It just skips them as of now. Do you mean we should set the SKIP_HOST_POLICY based on some argument passed to script rather than depending on -enableHostPolicy flag?

oneiro-naut replied:
I am going to add one argument called -testHostPolicy to both test-scenarios-in-runtime.sh and test-scenarios-local.sh.

oneiro-naut committed:
Add test script arguments and fix test skip conditions

This commit adds 4 options to run the test scripts.

Test script commands now become:
 ./test-scenarios-local.sh [OPTION] [ARMOR_OPTIONS]
 ./test-scenarios-in-runtime.sh [OPTION]

[OPTION]: [-testHostPolicy|-testContainerPolicy|-testAllButNative|-testAll]

Fixes: #168

Signed-off-by: Ayush Dwivedi <ayush.dwivedi@accuknox.com>

Commit:
Update test-scenarios-in-runtime.sh

Commit:
Update test-scenarios-local.sh

Commit:
Update test-scenarios-in-runtime.sh

Commit:
Update test-scenarios-local.sh
